backend/services/deployments/server.go

```diff
@@ -221,5 +225,26 @@ func RunServer(ctx context.Context) error {
 		return http.ListenAndServeTLS(listen, cert, key, handler)
 	}
 
-	return http.ListenAndServe(listen, handler)
+	srv := &http.Server{
+		Addr:    listen,
+		Handler: handler,
+	}
+
+	go func() {
+		if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			l.Fatalf("listen: %s\n", err)
+		}
+	}()
+
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, unix.SIGINT, unix.SIGTERM)
+	<-quit
+
+	ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := srv.Shutdown(ctxWithTimeout); err != nil {
+		l.Fatal("error when shutting down the server ", err)
```

Comment:
Can we log the error to the `Error` stream and pass the error to the parent function?
```suggestion
		l.Errorf("error when shutting down the server: %s", err.Error())
		return err
```